python/lance/data/convert/base.py

```diff
@@ -106,16 +106,20 @@ def _convert_field(self, name, typ, col):
         """pyarrow is unable to convert ExtensionTypes properly in pa.Table.from_pandas"""
         if isinstance(typ, pa.ExtensionType):
             storage = pa.array(col, type=typ.storage_type)
-            arr = pa.ExtensionArray.from_storage(typ, storage)
+            return pa.ExtensionArray.from_storage(typ, storage)
         elif pa.types.is_list(typ):
             native_arr = pa.array(col)
+            if isinstance(native_arr, pa.NullArray):
+                return pa.nulls(len(native_arr), typ)
             offsets = native_arr.offsets
             values = native_arr.values.to_numpy(zero_copy_only=False)
             return pa.ListArray.from_arrays(
                 offsets, self._convert_field(f"{name}.elements", typ.value_type, values)
             )
         elif pa.types.is_struct(typ):
             native_arr = pa.array(col)
+            if isinstance(native_arr, pa.NullArray):
```

Comment:
Assuming this was a fix for the demo notebook?

Could you add a test please?

Reply:
added